roles/scm/git/tasks/commit.yml

```diff
@@ -0,0 +1,62 @@
+---
+
+- name: Pre-flight checks
+  assert:
+    that: "{{ assertion.that }}"
+    fail_msg: "{{ assertion.msg }}"
+    quiet: true
+  loop_control:
+    loop_var: assertion
+  loop:
+    - msg: Repository Path variable must be set in 'scm_dir'
+      that:
+        - scm_dir is defined
+
+- name: Add all new files to repository
+  shell: git add --all
+  args:
+    chdir: "{{ scm_dir }}"
+  changed_when: false
+
+- name: Show staged files
+  shell: git diff --cached --name-only
+  args:
+    chdir: "{{ scm_dir }}"
+  register: scm_files_staged
+  changed_when: false
+
+- name: Configure git username
+  shell: git config user.name "{{ git.username }}"
```

Comment:
can we please add the `--local` flag here to avoid accidentally overwriting someone's global configuration (yes, it's potentially a corner case / error leg, but better to be safe...)

Applies to all `git config` commands